src/Compilers/Core/MSBuildTaskTests/CscTests.cs

```diff
@@ -27,6 +27,15 @@ public void MultipleSourceFiles()
             Assert.Equal("/out:test1.exe test1.cs test2.cs", csc.GenerateResponseFileContents());
         }
 
+        [Fact] 
+        public void DeterministicFlag()
+        {
+            var csc = new Csc();
+            csc.Sources = MSBuildUtil.CreateTaskItems("test.cs");
+            csc.Deterministic = true;
+            Assert.Equal("/deterministic+ /out:test.exe test.cs", csc.GenerateResponseFileContents());
+        }
```

Comment:
Would that be a part of the MSBuild tests though?  Or are you talking about the CSC command line options?

Since I implemented it the same way, I assume that `/deterministic+ /deterministic-` would be equivalent to `/optimize+ /optimize-`.